src/tooling/grain_doc.md

```diff
@@ -0,0 +1,144 @@
+
+## Attributes
+
+### @param
+
+```gr
+/**
+ * @param name: description
+ */
+```
+The `@param` attribute indicates a parameter to the function that the docblock is documenting.
+
+These should be specified in order of the function parameters and the name specified before the colon (`:`) should match the parameter name exactly—while not currently enforced, it will be required for named parameters in the future. The content after the colon is the description of the parameter.
+
+When generating docs, the type of the parameter will be detected by the compiler and inserted into the "Parameters" table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` attribute indicates the value that is returned by the function that the docblock is documenting.
+
+These must be provided a description of the return value. In the Grain stdlib, we don't use the `@returns` attribute if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and inserted into the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` attribute can be added to show how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and inserted into an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` attribute can be used to specify when the documented code was added.
+
+This must be provided a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next" instead of the version.
+
+### @history
+
+```gr
+/**
+ * @history version: description
+ */
+```
+
+The `@history` attribute can be used to share significant changes to the documented code.
+
+This must be provided a [Semantic Version](https://semver.org/), optionally prefixed with `v` before the colon (`:`) and a description of the change after the colon.
+
+When generating docs, will add a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
```

Comment:
```suggestion
When generating docs, adds a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
```